api/android/api/src/main/java/org/nnsuite/nnstreamer/TensorsData.java

```diff
@@ -33,63 +34,74 @@
      * @param size The byte size of the buffer
      *
      * @return The new byte buffer
+     *
+     * @throws IllegalArgumentException if given size is invalid
      */
     public static ByteBuffer allocateByteBuffer(int size) {
```

Comment:
Is this API really needed as public? I think its better to just allocate allocate on the basis of tensorInfo. Also the returned buffer has no association with the TensorsData object..

Reply:
For the convenience to pass the buffer to native code. ByteBuffer in java is big-endian and non-direct buffer. It cannot be passed and used in native code.

Reply:
got it, thanks for the explanation.